### ISSUE TITLE
Feature request: Allow datepicker to select multiple days and return them as List<DateTime>

### ISSUE THREAD

**Issue by brilliant-ember** [flutter/flutter#22621]
The Current datepicker only supports selecting one day.
When  showDatePicker(params) is called it returns a DateTime for the selected date, which is great, but it will be even better if there is an option that allows for multiple date selection and returning them as a list of <DateTime>.
This is not an actual issue but a feature that would make flutter shine more.

**Comment by jslavitz** [comment]
@brilliant-ember – Do you have an idea of what the UX would look like for this? In the meantime maybe you could have a button next to the picker that adds the currently selected date to a list and allows for the user to pick a new date?

**Comment by brilliant-ember** [comment]
@jslavitz -- For the UX I think something similar to AirBnb's date picker would be good, in a nutshell flutter's built in datepicker will de-select the currently selected date when you press on a new one, having a boolean option allows the date picker to retain the last selected while selecting a new date.
For example you have a blue circle on the selected date, if the user selected another date both dates now should have the circle around them

**Comment by reignierleo** [comment]
Hey, I would also love this feature. Anything new here?

**Comment by ElasticBottle** [comment]
@reignierleo I think the community have answered with the calenderoo package. Haven't used it myself, but it might be what you are looking for! 

https://pub.dev/packages/calendarro

**Comment by ram231** [comment]
Bump! https://material.io/components/pickers#mobile-pickers

Although there is a package by syncfusion. But skeptical on its license. :\
https://pub.dev/packages/syncfusion_flutter_calendar

**Comment by abhijith-vm** [comment]
> Hey, I would also love this feature. Anything new here?

yes i have a Usecase of selecting 5 dates from calendar widget. this feature would be super fine

**Comment by abhijith-vm** [comment]
> @reignierleo I think the community have answered with the calenderoo package. Haven't used it myself, but it might be what you are looking for!
> 
> https://pub.dev/packages/calendarro

thanx mahnn !!!! you are a  life saver

**Comment by enricobenedos** [comment]
Can it be considered close due to the Flutter 1.20.0 ` await showDateRangePicker();` implementation?
